relayer/src/chain/cosmos.rs

```diff
@@ -638,16 +638,44 @@ impl ChainEndpoint for CosmosSdkChain {
             .map_err(|_| Error::ics02(ClientError::empty_prefix()))
     }
 
-    /// Query the chain status
-    fn query_status(&self) -> Result<ChainStatus, Error> {
-        crate::time!("query_status");
-        crate::telemetry!(query, self.id(), "query_status");
+    /// Query the application status
+    fn query_application_status(&self) -> Result<ChainStatus, Error> {
+        crate::time!("query_application_status");
+        crate::telemetry!(query, self.id(), "query_application_status");
 
-        self.rt.block_on(query_status(
-            self.id(),
-            &self.rpc_client,
-            &self.config.rpc_addr,
-        ))
+        // We cannot rely on `/status` endpoint to provide details about the latest block.
+        // Instead, we need to pull block height via `/abci_info` and then fetch block
+        // metadata at the given height via `/blockchain` endpoint.
+        let abci_info = self
+            .block_on(self.rpc_client.abci_info())
+            .map_err(|e| Error::rpc(self.config.rpc_addr.clone(), e))?;
+
+        // Query `/blockchain` endpoint to pull the block metadata corresponding to
+        // the latest block that the application committed.
+        // TODO: Replace this query with `/header`, once it's available.
+        //  https://github.com/informalsystems/tendermint-rs/pull/1101
+        let blocks = self
+            .block_on(
+                self.rpc_client
+                    .blockchain(abci_info.last_block_height, abci_info.last_block_height),
```

Comment:
Note to reviewers:
- we query `/blockchain?minHeight=last_block_height&maxHeight=last_block_height` to fetch the metadata for a single block